### PR TITLE
Make the forced_update of @cached and @memoize accept the same parameters as the decorated function

### LINKED ISSUE #112: forced_update parameter of cached and memoized might receive the some parameters as the function
Maybe i just canʼt see some obvious way, but how the `forced_update` callable is supposed to decide if the value should be refreshed in the cache?

My idea was to pass a `refresh_cache=True` parameter to my function and use that in this callable. However, neither `@memoize` nor `@cached` pass any arguments to `forced_update`.

Is this intentional? I know it would be a breaking change to change this behaviour, though.

### PR DESCRIPTION
**Pull request by gergelypolonkai**
Fixes #112

### CODE REVIEW

File: flask_caching/__init__.py [hunk 1]
```diff
@@ -46,6 +46,29 @@
     null_control = (dict((k, None) for k in delchars),)
 
 
+def wants_args(f):
+    """Check if the function wants any arguments
+    """
+
+    try:
```
Comment (sh4nks) [inline flask_caching/__init__.py]: Now that Python 2 is removed, I guess we can simplify this now?

File: flask_caching/__init__.py [hunk 2]
```diff
@@ -390,7 +413,7 @@ def decorated_function(*args, **kwargs):
                             args, kwargs, use_request=True
                         )
 
-                    if callable(forced_update) and forced_update() is True:
+                    if callable(forced_update) and (forced_update(*args, **kwargs) if wants_args(forced_update) else forced_update()) is True:
```
Comment (sh4nks) [inline flask_caching/__init__.py]: The line is a bit to long. I'd like to keep the line length to 80.

I am using [black](https://github.com/python/black) to keep my python files nicely formatted (we gotta add this to our not yet existing CONTRIBUTING file :P)

**Comment by gergelypolonkai** [comment]
I think i fixed everything we talked about. Also, finally we have a coverage increase! :smile:

File: tests/test_memoize.py [hunk 3]
```diff
@@ -460,6 +460,67 @@ def big_foo(a, b):
...
+
+        def forced_update_func(a, b):
+            call_counter[1] += 1
+            call_params[call_counter[1] - 1] = (a, b)
+
+            return forced_update
+
+        @cache.memoize(5, forced_update=forced_update_func)
+        def memoized_func(a, b):
+            print('memoizing')
```
Comment (sh4nks) [inline tests/test_memoize.py]: Is this print statement on purpose?

Comment (gergelypolonkai) [reply tests/test_memoize.py]: No, it wasn’t :blush:

**Pull request merged**